packages/node_modules/@webex/webex-core/src/lib/services/services.js

```diff
@@ -684,6 +688,74 @@ const Services = WebexPlugin.extend({
       .then(({body}) => this._formatReceivedHostmap(body));
   },
 
+  /**
+   * Initialize the discovery services and the whitelisted services.
+   *
+   * @returns {void}
+   */
+  initConfig() {
+    // Get the catalog and destructure the services config.
+    const catalog = this._getCatalog();
+    const {services} = this.webex.config;
+
+    // Validate that the services configuration exists.
+    if (services) {
+      // Check for discovery services.
+      if (services.discovery) {
+        // Format the discovery configuration into an injectable array.
+        const formattedDiscoveryServices = Object.keys(services.discovery)
+          .map((key) => ({
+            name: key,
+            defaultUrl: services.discovery[key]
+          }));
+
+        // Inject formatted services into services catalog.
+        catalog.updateServiceUrls('discovery', formattedDiscoveryServices);
+      }
+
+      // Check for allowed host domains.
+      if (services.allowedDomains) {
+        // Store the allowed domains as a property of the catalog.
+        catalog.setAllowedDomains(services.allowedDomains);
+      }
+
+      // Set `validateDomains` property to match configuration
+      this.validateDomains = services.validateDomains;
+    }
+  },
+
+  /**
+   * Make the initial requests to collect the root catalogs.
+   *
+   * @returns {Promise<void, Error>} - Errors if the token is unavailable.
+   */
+  initServiceCatalogs() {
+    this.logger.info('services: initializing initial service catalogs');
+
+    // Destructure the credentials plugin.
+    const {credentials} = this.webex;
+
+    // Init a promise chain to catch errors.
+    return Promise.resolve()
+      // Get the user's OrgId.
+      .then(() => credentials.getOrgId())
+      // Begin collecting the preauth/limited catalog.
+      .then((orgId) => this.collectPreauthCatalog({orgId}))
+      .then(() => {
+        // Validate if the token is authorized.
+        if (credentials.canAuthorize) {
+          // Attempt to collect the postauth catalog.
+          return this.updateServices()
+            .catch(() => this.logger.warn(
```

Comment:
What else happens if the services can't be updated? could the client still proceed with whatever it had?

Reply:
Yes, if the token is not a fully authorized user token, it will not proceed with just the preauth catalog. This works in the cases of client auth, in which a user token may not be present at the time of `webex:ready`